### PR TITLE
fix(lint): no-html-link in appDir

### LINKED ISSUE #51742: @next/next/no-html-link-for-pages doesn't work with app router
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 22.5.0: Mon Apr 24 20:52:24 PDT 2023; root:xnu-8796.121.2~5/RELEASE_ARM64_T6000
    Binaries:
      Node: 18.12.1
      npm: 8.19.2
      Yarn: 1.22.19
      pnpm: 8.6.2
    Relevant packages:
      next: 13.4.7
      eslint-config-next: 13.4.7
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.1.3
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

_No response_

### Link to the code that reproduces this issue or a replay of the bug

https://github.com/osdiab/next-eslint-repro-links/blob/main/src/app/page.js#L13

### To Reproduce

- set up next app
- make an `about` page
- add a link to the `index` page to the `/about` page with an `<a />` tag
- run `next lint`
- no error shows up

### Describe the Bug

I expect from [the description of the lint rule](https://nextjs.org/docs/messages/no-html-link-for-pages) that if i use an `<a />` tag where I should be using a `<Link />` tag for an internal link, it would tell me that i should use `<Link />` instead.

### Expected Behavior

It should show a lint error

### Which browser are you using? (if relevant)

_No response_

### How are you deploying your application? (if relevant)

_No response_

### PR DESCRIPTION
**Pull request by DerTimonius**
supersedes #51783 
## What
This PR fixes the issue of `<a>` tags for internal routing not being caught by linting in the app directory.

## Why
The linting of the html links should be consistent across pages and app directories.

## How
- Added a new function to parse the URLs in the appDir (since it's a bit different from the /pages directory)
- Lint now checks against pagesUrls and appUrls
- Added tests for the appDir in the no-html-link-for-pages test file

fixes #51742

### CODE REVIEW

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: 9b2e8b579234a2503ce123dd8ce6f2bac5c64c81

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Comment by DerTimonius** [comment]
@ijjk is there something missing to get the CI approved?

**Comment by DerTimonius** [comment]
@leerob maybe you can help in getting this reviewed?

File: packages/eslint-plugin-next/src/utils/url.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import * as path from 'path'
 import * as fs from 'fs'
 
+import { normalizeAppPath } from '../../../next/src/shared/lib/router/utils/app-paths'
```
Comment (ijjk) [inline packages/eslint-plugin-next/src/utils/url.ts:4]: We can't import relatively from the `next` package like this as they are published independently of each other. We would need to copy the logic in completely

**Review by ijjk — APPROVED** [review]

**Pull request merged**